main.tf

```diff
@@ -64,6 +64,15 @@ resource "aws_s3_bucket" "bucket" {
   }
 }
 
+# explicitly block public access
+resource "aws_s3_bucket_public_access_block" "bucket" {
+  bucket                  = aws_s3_bucket.bucket.id
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
```

Comment:
Note that turning this on will prevent any cross-account access to the bucket, even if explicitly defined in the bucket policy and restricted to a single other account. Not sure what use cases need cross-account access to tfstate, but might want to make it a configurable parameter instead of forcing true here.

Reply:
added a `block_public_access` variable which defaults to `true` but also won't create the `aws_s3_bucket_public_access_block` resource when set to `false`

Reply:
Looks good!